extension/src/popup/components/sendPayment/SendAmount/index.tsx

```diff
@@ -151,6 +151,36 @@ export const SendAmount = ({ previous }: { previous: ROUTES }) => {
     formik.values.amount,
   ]);
 
+  const getAmountFontSize = () => {
+    const length = formik.values.amount.length;
+    if (length <= 9) {
+      return "";
+    }
+    if (length <= 15) {
+      return "med";
+    }
+    return "small";
+  };
+
+  // remove non digits and decimal
+  const cleanAmount = (s: string) => s.replace(/[^0-9.]/g, "");
+
```

Comment:
2 formatAmount notes:
- after 16 digits Number() will round, so maxing the pre-decimal digits to 16
- unfortunately we can't just do decimal.format(Number(val)), because it removes the decimal if no digits afterward (ie `12.` becomes `12`, so need to split it)